### PR TITLE
Fixed #30890 -- Added MariaDB support for the relate lookup.

### PR DESCRIPTION
**Pull request by felixxm**
https://code.djangoproject.com/ticket/30890

### CODE REVIEW

File: tests/gis_tests/geoapp/tests.py [hunk 1]
```diff
@@ -461,7 +461,11 @@ def test_relate_lookup(self):
             intersects_mask = 'overlapbdyintersect'
 
         # Testing contains relation mask.
-        self.assertEqual('Texas', Country.objects.get(mpoly__relate=(pnt1, contains_mask)).name)
+        if connection.features.supports_transform:
+            self.assertEqual(
+                Country.objects.get(mpoly__relate=(pnt1, contains_mask)).name,
+                'Texas',
+            )
```
Comment (felixxm) [inline tests/gis_tests/geoapp/tests.py]: MariaDB doesn't support the `Transform` function so `geom` must have the same SRID.

File: docs/releases/3.1.txt [hunk 2]
```diff
@@ -59,7 +59,7 @@ Minor features
 :mod:`django.contrib.gis`
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* ...
+* Added MariaDB support for the :lookup:`relate` lookup.
```
Comment (carltongibson) [inline docs/releases/3.1.txt]: Present tense. 

```suggestion
* MariaDB now supports the :lookup:`relate` lookup.
```

(or such)?

File: tests/gis_tests/geoapp/tests.py [hunk 3]
```diff
@@ -450,7 +450,7 @@ def test_relate_lookup(self):
                 qs.count()
 
         # Relate works differently for the different backends.
-        if postgis or spatialite:
+        if postgis or spatialite or (mysql and connection.mysql_is_mariadb):
```
Comment (carltongibson) [inline tests/gis_tests/geoapp/tests.py]: This isn't very pretty. 

Is it worth adding a `mariadb` shortcut boolean to `../utils.py`?

**Comment by felixxm** [comment]
@carltongibson Thanks :+1: updated.

**Pull request merged**